source/auth/auth.rst

```diff
@@ -250,6 +250,48 @@ or "SCRAM-SHA-256").
 The cache entry value MUST be either the ``saltedPassword`` parameter or the
 combination of the ``clientKey`` and ``serverKey`` parameters.
 
+Speculative Authentication via Handshake
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:since: 4.4
+
+The ``isMaster`` handshake supports a new argument, ``speculativeSaslStart``,
+provided as a BSON object. Clients specifying this argument to ``isMaster`` will
+speculatively include the first ``saslStart`` command of an authentication handshake.
+This command may be provided to the server in parallel with any standard request for
+supported authentication mechanisms. This would permit clients to merge their
+``saslStart`` message with their ``isMaster`` request, and receive the ``saslStart``
+reply with the ``isMaster`` reply.
+
+If the speculatively issued command succeeds, the client should proceed with the next
+step of the exchange. If the command fails, the client should determine whether the
+``isMaster`` response contains enough information to select another mechanism.
```

Comment:
>  the client should determine whether the `isMaster` response contains enough information to select another mechanism.

Can we make this specific?  Is this about `saslSupportedMechs` in the response?

Reply:
I have reworded this section based on the new `speculativeAuthenticate` argument.